kubeconfig.go

```diff
@@ -0,0 +1,69 @@
+package kubeconfig
+
+import (
+	"context"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewG8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (versioned.Interface, error){
```

Comment:
```suggestion
func (k KubeConfig) NewG8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (versioned.Interface, error){
```

As this is a library these needs to be methods rather than functions. We also need to define an interface in spec.go. See https://github.com/giantswarm/tenantcluster/blob/master/spec.go for an example.

Also godoc is needed here and in spec.go.

Reply:
Why do we need these to be methods?

Reply:
I think the k8s client should be passed in. Currently in cluster is used but that won't work when running outside k8s.

Reply:
Yes, that makes sense. 👍

Reply:
btw, do we need to stick with the g8s naming here? everywhere else we already removed/replaced it

Reply:
@rossf7 So I think, what you mean is that I should make `New` function which would take `k8sclient` as a parameter so we can use it instead of `InClusterConfig`?
https://github.com/giantswarm/kubeconfig/blob/3f345bc53fa3b310cd291e64bebfff02b907e266/kubeconfig.go#L42

Reply:
> k8sclient as a parameter so we can use it instead of InClusterConfig

@tomahawk28 Yes the k8s client should be passed in rather than using "in cluster".

This is more flexible. app-operator will be configured to use "in cluster" via its Helm chart. You can use that k8s client when configuring this library in app-operator.

Reply:
> btw, do we need to stick with the g8s naming here? everywhere else we already removed/replaced it

@puja108 I know G8s (AKA Giantnetes) is deprecated but the naming here is very useful. Makes it easily clear if this is a regular K8s client or a generated client from `apiextensions`. I wouldn't want to remove it unless we had a good replacement.